src/NextGen/Framework/FormExtensions/ServiceProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Give\NextGen\Framework\FormExtensions;
+
+use Give\Helpers\Hooks;
+use Give\NextGen\Framework\FormExtensions\Registrars\FormExtensionRegistrar;
+use Give\ServiceProviders\ServiceProvider as ServiceProviderInterface;
+
+/**
+ * @unreleased
+ */
+class ServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register()
+    {
+        give()->singleton(FormExtensionRegistrar::class, function () {
+            $registrar = new FormExtensionRegistrar();
+
+            Hooks::doAction('givewp_register_form_extension', $registrar);
```

Comment:
Running `do_action` inside of the `register()` method may be too soon in the setup process. Instead we should probably be triggering hooks in the `boot()` method.

Reply:
Changed it on 4a407d9d52e2d4c1591de3afc332b129c0b62272 then tested it. Still works fine.